poc-07.py

```diff
@@ -240,6 +240,32 @@ def run(self):
         print("- from left: %s"% leftMessages)
         print("- from rght: %s"% rightMessages)
 
+        for leftMessage in leftMessages:
+            for rightMessage in rightMessages:
+                if leftMessage.uid == rightMessage.uid:
+                    leftChanges = leftMessage.getChanges()
+                    rightChanges = rightMessage.getChanges()
+                    changes = {'read': None, 'important': None}
+
+                    #FIXME: fix conflicts
+                    # We are here assuming that leftChanges and rightChanges
+                    # will never create conflict situation. In other words
+                    # for a particular flag either leftChanges or
+                    # rightChanges will provide value for it.
+                    for flag, value in leftChanges.items():
+                        if value != None:
+                            changes[flag] = value
+
+                    for flag, value in rightChanges.items():
+                        if value != None:
+                            changes[flag] = value
+
+                    for flag, value in changes.items():
+                        if value != None:
+                            leftMessage.mark(flag, value)
+                            rightMessage.mark(flag, value)
+
+
```

Comment:
We are working on collections of `Messages`. Wouldn't it worth introducing a new `Messages` method like `Messages.merge(messages)`?